design-documents/graph-ql/extensions.md

```diff
@@ -0,0 +1,112 @@
+# GraphQL Extensions
+
+Extension developers need a way to extend our GraphQL API. This document is a proposal of how this can be done in the context of a _Node.js_ GraphQL API, should we decide to go in that direction.
+
+## Package registry
+
+First, we need a place to publish both public and private JavaScript extensions and in a secure way. At the time of this writing, [npm][] is the de facto package registry for JavaScript; however, the [GitHub Package Registry][] is currently in beta and offers the following features:
+
+- Safely publish and consume packages within your organization or with the entire world.
+- Publish privately for your team or publicly for the open source community.
+- Discover and publish public and private packages in one place.
+- Seamlessly use and reuse any package as a dependency in a project by downloading it straight from GitHub.
+
+As merchants currently have the ability to install public extensions or purchase private ones, having them both published all in one place is extremely beneficial for our needs.
+
+_This topic needs further investigation after the official release, which is expected to be during the [GitHub Universe](https://githubuniverse.com/) event, November 13-14, 2019._
```

Comment:
In addition we need to understand how to integrate JS extension distribution with Magento Marketplace. Please discuss this question with @ravmenon